text/0009-golang.md

```diff
@@ -0,0 +1,70 @@
+
+Thanks to its design, Go has a high level of familiarity for Java (and C#) developers. This includes both the language syntax and programming constructs, as well as the runtime characteristics. As previously mentioned, like Java, Go is a statically typed, compiled language with a garbage collector. While Go redefines some basic concepts like loops or conditionals (e.g there's no while loop in Go and the switch statement does not fall through) there are few paradigm-shifting concepts, with the concurrency model (there's no direct equivalent of Go's goroutines in Java) and more control over memory management (pointers and stack vs heap allocation) being some examples where the languages differ substantially.  This familiarity means that a Java programmer can become proficient with the Go language relatively quickly, for instance through the duration of a small development project, including idiomatic use of some of the constructs that differ from Java. Similarity to Java also means that the developer can “read” Go source code with limited prior knowledge of the language, referring only occasionally to the language documentation. 
+ 
+The simplicity of Go stems from the fact that, unlike Java, it is not a multi-purpose tool. Go's main uses cases are web services and CLIs (command-line interfaces) which makes it a perfect fit for writing FOLIO backend modules. Being simpler than Java, the language is also easier to pick up by full-stack developers coming from scripting and web languages background like Node.js or Python. This can be a big advantage for small teams of few developers as the entire team can, effectively, be kept full-stack. With large Java codebases it’s not practical and developers in the team usually specialize in front-end or back-end programming. 
+
+On the operational front, Go's characteristics make it a better fit than Java for building systems comprised of many small microservices. Go is compiled ahead-of-time (AOT) into a static binary which is a perfect fit for Docker containers. AOT compilation means that the resulting application has a visibly smaller memory footprint and faster startup times as compared to Java/vert.x, and orders of magnitude better as compared to Java/Spring based modules. Java provides AOT compilation capability through the GraalVM project, but porting existing Java projects to GraalVM involves substantial complexity and is not possible for certain frameworks like older Spring (including versions used by FOLIO project) and Grails due to the dynamic language features they require. 
+
+From a sysops perspective, Go containers require less operational tuning compared to the JVM containers as they do not contain a stand-alone virtual machine that requires parametrization.
```

Comment:
It's important to note that "less" does not mean "none".  My understanding is that there are still several knobs which may need to be adjusted, e.g. memory limits, garbage collector, etc.